cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -239,7 +239,7 @@ std::vector<std::future<ResultT>> make_sliced_tasks(
   std::vector<std::future<ResultT>> slice_tasks;
   std::transform(slices.cbegin(), slices.cend(), std::back_inserter(slice_tasks), [&](auto& slice) {
     return pool.submit_task(
-      [&] { return function(ptr + slice.offset, slice.size, offset + slice.offset); });
+      [=] { return function(ptr + slice.offset, slice.size, offset + slice.offset); });
```

Comment:
I don't understand how this worked for as long as it did.

Reply:
It is similar to the bug that I recently fixed for ORC reader, when reading file using the default stream `0`. It's never showed up until recently.